### ISSUE TITLE
Handling cash-settled options

### ISSUE THREAD

**Issue by johngriffin** [alexpung/UK-Investment-tax-calculator#109]
Hi Alex

Great project you've created here, thanks for putting it together.

I've just uploaded my data and I think i've found an issue.  Once all the data is parsed correctly and I click the "Start Calculation" button I get the following error:

```
Error
No corresponding OptionAssigned trade found for option (Underlying: XSP, Quantity: 100, date: 07/11/2023 00:00:00, there is likely an omission of trade(s) in the input)
```

I suspect this is because XSP is cash-settled and therefore does not get assigned if held until expiry.

I would normally look to create a pull request but I'm a python guy and have no idea where to start here so kicking it back to you - sorry!

**Comment by johngriffin** [comment]
Let me know if you'd like the trade data that created the error, happy to share that privately

**Comment by alexpung** [comment]
Can you check in your files if you have statement of fund for option settlement? It should be like these below.

<root>
                <StatementOfFundsLine assetCategory='OPT' symbol='SPX 180316C01000000' 
                    activityDescription='Option Cash Settlement for: Assignment' 
                    reportDate='16-Mar-18' amount='-700732' currency='USD' />
                <StatementOfFundsLine assetCategory='STK' symbol='AAPL' 
                    activityDescription='Option Cash Settlement for: Exercise' 
                    reportDate='16-Mar-18' amount='5000' currency='USD' />
                <StatementOfFundsLine assetCategory='OPT' symbol='SPX 180316P01000000' 
                    activityDescription='Option Cash Settlement for: Exercise' 
                    reportDate='16-Mar-18' amount='-12345' currency='USD' />
            </root>

**Comment by johngriffin** [comment]
If I search for "cash settlement" I can find only one line:

```
<StatementOfFundsLine accountId="------" acctAlias="" model="" currency="USD" fxRateToBase="1" assetCategory="OPT" subCategory="C" symbol="XSP   231107C00437000" description="XSP 07NOV23 437 C" conid="659181273" securityID="" securityIDType="" cusip="" isin="" figi="BBG01JPN1FS4" listingExchange="CBOE" underlyingConid="137851301" underlyingSymbol="XSP" underlyingSecurityID="" underlyingListingExchange="" issuer="" issuerCountryCode="" multiplier="100" strike="437" expiry="07-Nov-23" putCall="C" principalAdjustFactor="" reportDate="08-Nov-23" date="07-Nov-23" settleDate="08-Nov-23" activityCode="ADJ" activityDescription="Option Cash Settlement for: Assignment ( XSP 07NOV23 437 C )" tradeID="" relatedTradeID="" orderID="" buySell="" tradeQuantity="0" tradePrice="0" tradeGross="0" tradeCommission="0" tradeTax="0" debit="-84" credit="" amount="-84" tradeCode="" balance="94307.448622182" levelOfDetail="BaseCurrency" transactionID="25549488950" origTransactionID="" relatedTransactionID="" actionID="" serialNumber="" deliveryType="" commodityType="" fineness="0.0" weight="0.0" />

```
Which seems strange because there are more times that XSP options are purchased in the data.  I would expect that it's not the only time XSP options were held to expiry in this tax year but I'm not sure how to easily check that.

**Comment by alexpung** [comment]
Likely because I use the reportDate which doesn't match the trade date
`date="07-Nov-23" settleDate="08-Nov-23"`

This section of code get run here and you don't have a match.
Please try again after the last PR get deployed.

             else
            {
                var matchingCashSettlement = tradeList.CashSettlements.Find(trade => trade.AssetName == optionTrade.AssetName &&
                                                                                     trade.Date.Date == optionTrade.Date.Date &&
                                                                                     trade.TradeReason == optionTrade.TradeReason);
                if (matchingCashSettlement is not null)
                {
                    optionTrade.CashSettled = true;
                    WrappedMoney tradeValue;
                    if (matchingCashSettlement.TradeReason == TradeReason.OptionAssigned) tradeValue = matchingCashSettlement.Amount * -1;
                    else tradeValue = matchingCashSettlement.Amount;
                    optionTrade.GrossProceed = optionTrade.GrossProceed with { Amount = tradeValue, Description = matchingCashSettlement.Description };
                }
                else
                {
                    toastService.ShowError($"No corresponding {optionTrade.TradeReason} trade found for option (Underlying: {optionTrade.Underlying}, " +
                    $"Quantity: {optionTrade.Quantity * optionTrade.Multiplier}, date: {optionTrade.Date.Date}, there is likely an omission of trade(s) in the input)");
                }

**Comment by johngriffin** [comment]
Awesome!  Looks like that has fixed it.

Thanks :)

**Comment by alexpung** [comment]
Fixed in #110 #111
